### DISCUSSION TITLE
Variant types from RecipeVariants doesn't work

### DISCUSSION THREAD

**Discussion by gabrieloureiro** [vanilla-extract-css/vanilla-extract#1035] [Q&A] [answered]
I've this occurrence when i call this property inside the component. This is an issue?

1) styles: 
![Screen Shot 2023-03-09 at 12 54 34](https://user-images.githubusercontent.com/39869298/224079698-bb5d2728-280c-46ff-a2a9-6bcb69f502b7.png)

2) component: 
![Screen Shot 2023-03-09 at 12 58 19](https://user-images.githubusercontent.com/39869298/224080749-0350463d-e3db-4293-ab4b-b212f1760102.png)

**Comment by askoufis** [comment, marked as answer]
Your `TextVariantSize` type isn't what you think it is. It's an object containing a property `size` that can accept all your size variants `xxs`, `xs`, etc., rather than just a union of all your size variants.

I think to fix this you'll want to extract the variants from the recipe type:

```ts
type TextVariantSize = Required<NonNullable<RecipeVariants<typeof textStyles>>>['size'];
```

Alternatively, you could construct your `TextProps` type in a different way:

```ts
Atoms & { as?: AllowedHtmlTags } & RecipeVariants<typeof textStyles>
```

**Reply by gabrieloureiro** [reply]
Thank you @askoufis. All typed and working as well.

**Reply by askoufis** [reply]
Great! Glad I could help.
